### PR TITLE
fix(CX-1991): new onboarding UI changes 

### PR DESCRIPTION
**Pull request by MounirDhahri**
The type of this PR is: **Bugfix**

This PR resolves [CX-1991]

### Description
- New Onboarding UI Changes (mostly copy and size changes)
- Add option to Google auth 

https://user-images.githubusercontent.com/11945712/135080865-7b609ec4-18dd-4dd6-a4de-992f2e8281c8.mov

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Add Google Auth - mounir
- Update Onboarding UI - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[CX-1991]: https://artsyproduct.atlassian.net/browse/CX-1991

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 16766061e19e19bb212ee44e42f5938bb78c1993
</p>

File: src/lib/Scenes/Onboarding/OnboardingCreateAccount/OnboardingCreateAccountEmail.tsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ export const OnboardingCreateAccountEmail: React.FC<OnboardingCreateAccountEmail
   return (
     <OnboardingCreateAccountScreenWrapper
       onBackButtonPress={route.params.navigateToWelcomeScreen}
-      title="Sign up with email"
+      title="Sign Up with Email"
```
Comment (lordkiz) [inline src/lib/Scenes/Onboarding/OnboardingCreateAccount/OnboardingCreateAccountEmail.tsx:27]:
```suggestion
      title="Sign Up With Email"
```

File: src/lib/Scenes/Onboarding/OnboardingCreateAccount/OnboardingCreateAccountPassword.tsx [hunk 2]
```diff
@@ -19,7 +19,7 @@ export const OnboardingCreateAccountPassword: React.FC<OnboardingCreateAccountPa
   return (
     <OnboardingCreateAccountScreenWrapper
       onBackButtonPress={navigation.goBack}
-      title="Create a password"
+      title="Create a Password"
```
Comment (lordkiz) [inline src/lib/Scenes/Onboarding/OnboardingCreateAccount/OnboardingCreateAccountPassword.tsx:22]:
```suggestion
      title="Create A Password"
```

File: src/lib/Scenes/Onboarding/OnboardingLogin.tsx [hunk 3]
```diff
@@ -75,7 +75,7 @@ export const OnboardingLoginWithEmailForm: React.FC<OnboardingLoginProps> = ({ n
         keyboardShouldPersistTaps="always"
       >
         <Spacer mt={60} />
-        <Text variant="lg">Log in with email</Text>
+        <Text variant="lg">Login with Email</Text>
```
Comment (lordkiz) [inline src/lib/Scenes/Onboarding/OnboardingLogin.tsx:78]:
```suggestion
        <Text variant="lg">Login With Email</Text>
```

File: src/lib/Scenes/Onboarding/OnboardingPersonalization/OnboardingPersonalization.tsx [hunk 4]
```diff
@@ -62,9 +62,9 @@ const OnboardingPersonalizationListHeader = ({ navigateToModal }: { navigateToMo
   return (
     <>
       <Box px={2}>
-        <Text variant="lg">What artists do you collect?</Text>
+        <Text variant="lg">What Artists do You Collect?</Text>
```
Comment (lordkiz) [inline src/lib/Scenes/Onboarding/OnboardingPersonalization/OnboardingPersonalization.tsx:65]:
```suggestion
        <Text variant="lg">What Artists Do You Collect?</Text>
```

File: src/lib/Scenes/Onboarding/OnboardingSocialPick.tsx [hunk 5]
```diff
@@ -44,6 +47,20 @@ export const OnboardingSocialPick: React.FC<OnboardingSocialPickProps> = ({ mode
...
+  const useGoogle = async () => {
+    try {
+      if (mode === "login") {
+        await GlobalStore.actions.auth.authGoogle({ signInOrUp: "signIn" })
+      } else {
+        await GlobalStore.actions.auth.authGoogle({ signInOrUp: "signUp", agreedToReceiveEmails: true })
+      }
+    } catch (error) {
+      if (typeof error === "string") {
+        Alert.alert("Try again", error)
```
Comment (lordkiz) [inline src/lib/Scenes/Onboarding/OnboardingSocialPick.tsx:59]: Shouldn't we add action buttons to this alert?

File: src/lib/Scenes/Onboarding/OnboardingSocialPick.tsx [hunk 6]
```diff
@@ -55,106 +72,128 @@ export const OnboardingSocialPick: React.FC<OnboardingSocialPickProps> = ({ mode
...
       <BackButton
         onPress={() => {
           navigation.goBack()
         }}
       />
-      <Join separator={<Spacer height={60} />}>
-        <Text variant="lg">{mode === "login" ? "Log in" : "Create account"}</Text>
+      <Flex px={1.5}>
+        <Join separator={<Spacer height={60} />}>
+          <Text variant="xxl">{mode === "login" ? "Log in" : "Create account"}</Text>
```
Comment (lordkiz) [inline src/lib/Scenes/Onboarding/OnboardingSocialPick.tsx:83]:
```suggestion
          <Text variant="xxl">{mode === "login" ? "Log In" : "Create Account"}</Text>
```
I suppose we are following capitalize={'words'} approach for everything.

File: src/lib/Scenes/Onboarding/OnboardingSocialPick.tsx [hunk 7]
```diff
@@ -55,106 +72,128 @@ export const OnboardingSocialPick: React.FC<OnboardingSocialPickProps> = ({ mode
...
+                }
+              }}
+              block
+              haptic="impactMedium"
+              mb={1}
+              variant="outline"
+              icon={<EnvelopeIcon mr={1} />}
+              testID="useEmail"
+            >
+              {mode === "login" ? "Continue with email" : "Sign up with email"}
```
Comment (lordkiz) [inline src/lib/Scenes/Onboarding/OnboardingSocialPick.tsx:101]:
```suggestion
              {mode === "login" ? "Continue With Email" : "Sign Up With Email"}
```

File: src/lib/Scenes/Onboarding/OnboardingSocialPick.tsx [hunk 8]
```diff
@@ -55,106 +72,128 @@ export const OnboardingSocialPick: React.FC<OnboardingSocialPickProps> = ({ mode
...
+              <Button
+                onPress={useApple}
+                block
+                haptic="impactMedium"
+                mb={1}
+                variant="fillDark"
+                icon={<Image source={require("@images/apple.webp")} resizeMode="contain" style={{ marginRight: 10 }} />}
+                testID="useApple"
+              >
+                {mode === "login" ? "Continue with Apple" : "Sign up with Apple"}
```
Comment (lordkiz) [inline src/lib/Scenes/Onboarding/OnboardingSocialPick.tsx:113]:
```suggestion
                {mode === "login" ? "Continue With Apple" : "Sign Up With Apple"}
```

File: src/lib/Scenes/Onboarding/OnboardingSocialPick.tsx [hunk 9]
```diff
@@ -55,106 +72,128 @@ export const OnboardingSocialPick: React.FC<OnboardingSocialPickProps> = ({ mode
...
+                block
+                haptic="impactMedium"
+                mb={1}
+                variant="outline"
+                icon={
+                  <Image source={require("@images/google.webp")} resizeMode="contain" style={{ marginRight: 10 }} />
+                }
+                testID="useGoogle"
+              >
+                {mode === "login" ? "Continue with Google" : "Sign up with Google"}
```
Comment (lordkiz) [inline src/lib/Scenes/Onboarding/OnboardingSocialPick.tsx:128]:
```suggestion
                {mode === "login" ? "Continue With Google" : "Sign Up With Google"}
```

File: src/lib/Scenes/Onboarding/OnboardingSocialPick.tsx (hunk 5 above)
Comment (MounirDhahri) [reply src/lib/Scenes/Onboarding/OnboardingSocialPick.tsx:59]: So far, same as for the other auth methods, we have been using the default dismiss/ok alert that we get from the platform and in order for the user to try again, they will have to click that button to dismiss the modal.

Comment (lordkiz) [reply src/lib/Scenes/Onboarding/OnboardingSocialPick.tsx:59]: Ok. I was looking at the title that says "Try Again". One can easily assume that by pressing Ok the login is retried. This should not block merging though

**Comment by lordkiz** [comment]
> @lordkiz I followed the specs for the copy where we are title casing words except the ones specified [here](https://github.com/artsy/to-title-case/blob/a0f355185a165e18f763ef654a8fbaf25c3a5a4a/src/index.ts#L17) (the list is missing `with`)

Oh. Thanks for clarifying. @MounirDhahri

**Review by lordkiz — APPROVED** [review]
Awesome work! 👍 🎉

**Pull request merged**